collect/retryLimiter.go

```diff
@@ -0,0 +1,39 @@
+package collect
+
+import "sync"
+
+type retryLimiter struct {
+	limit int
```

Comment:
If we're using this, I am thinking the limit needs a lock -- or better yet, we could use the size of a channel as the lock. Fill up a channel, to acquire something you remove an item from the channel, and you put it back into the channel when you're done with it. You'll never block on adding something, but you might block on getting one.